src/main/riddl/Venues/venueMessages.riddl

```diff
@@ -0,0 +1,79 @@
+command EstablishVenue is {
+  venueId: VenueId briefly "A unique identifier for this Venue. This must be provided by the calling client.",
+  info: VenueContext.Info briefly "Details about a venue required for it to exist. See VenueContext.Info for details.",
+  locations: LocationId*,
+  establishingMember: MemberId briefly "The Id of the Member who is issuing the create command.",
+  owner: MemberId? briefly "Used in the case that establishingMember is not the owner"
+} briefly "Creates a Venue with all top-level fields, potentially without Locations"
+described in file "establishVenue.md"
+event VenueEstablished is {
+  venueId: VenueId,
+  info: VenueContext.Info,
+  locations: LocationId*,
+  meta: VenueContext.MetaInfo
+}
+
+command EditVenueInfo is {
+  venueId: VenueId briefly "The unique Id of the venue to create",
+  newInfo: VenueContext.UpdateInfo briefly "The information about the venue",
+  editingMember: MemberId briefly "Id of the member who is creating the venue"
+} briefly "A command to edit the values of an Venue's Info "
+described in file "editVenueInfo.md"
+event VenueInfoUpdated is {
+  venueId: VenueId,
+  info: VenueContext.Info,
+  meta: VenueContext.MetaInfo
+}
+
+command UpdateVenueStatus is {
+  venueId: VenueId,
+  status: VenueContext.Status,
+  updatingMember: MemberId
+}
+event VenueStatusUpdated is {
+  venueId: VenueId,
+  newStatus: VenueContext.Status
+}
+
+command AddVenueLocations is {
```

Comment:
I think VenueLocations should be maintained as a projection. The Location needs to have an attribute of VenueId to declare which venue it belongs to. I think the natural thought process would be, "I am going to add a new Location". When setting up that Location you would need to identify the Location it belongs to, if any. The LocationAdded/LocationRemoved events would update the projection.

Reply:
"LocationAdded/LocationRemoved" as in the creation events?

Reply:
Yeah, this comment was all kinds of confusing. Sorry. Location needs to have a VenueId (if there is one) to set up the relationship between Location and Venue. So when a Location is created we would update the project. Thinking about this for a minute more, the projection would only want to show Locations that were in the "Open" state. Draft and Closed Locations probably wouldn't be so helpful. Thoughts?

Reply:
Sure, I think that works. I mean we can always adjust the projection to include Draft and Closed if needed. One idea that comes to mind is caching the projection only for Open state. That may speed things up, without making the other data unavailable.

Reply:
I was suggesting what you observed in the last sentence. The projection only contains (caches) the Locations that are in Open state. So the projection is updated as Locations transition in and out of the Open state.